src/engine/Actor.ts

```diff
@@ -1,1128 +1,1126 @@
-       ctx.scale(this.scale.x, this.scale.y);
-       ctx.rotate(this.rotation);
-
-       // translate canvas by anchor offset
-       ctx.save();
-       ctx.translate(-(this._width * this.anchor.x), -(this._height * this.anchor.y));
-
-       this.emit('predraw', new PreDrawEvent(ctx, delta, this));
-                    
-       if (this.currentDrawing) {
-          var drawing = this.currentDrawing;
-          // See https://github.com/excaliburjs/Excalibur/pull/619 for discussion on this formula          
-          var offsetX = (this._width - drawing.naturalWidth * drawing.scale.x) * this.anchor.x;
-          var offsetY = (this._height - drawing.naturalHeight * drawing.scale.y) * this.anchor.y;
-
-          if (this._effectsDirty) {
+   }
+
+   /**
+    * Perform euler integration at the specified time step
+    */
+   public integrate(delta: number) {
+      // Update placements based on linear algebra
+      var seconds = delta / 1000;
+
+      var totalAcc = this.acc.clone();
+      // Only active vanilla actors are affected by global acceleration
+      if (this.collisionType === CollisionType.Active) {
+         totalAcc.addEqual(Physics.acc);
+      }
+
+      this.vel.addEqual(totalAcc.scale(seconds));
+      this.pos.addEqual(this.vel.scale(seconds)).addEqual(totalAcc.scale(0.5 * seconds * seconds));
+
+      this.rx += this.torque * (1.0 / this.moi) * seconds;
+      this.rotation += this.rx * seconds;
+
+      this.scale.x += this.sx * delta / 1000;
+      this.scale.y += this.sy * delta / 1000;
+
+      // Update physics body
+      this.body.update();
+   }
+
+   /**
+    * Called by the Engine, updates the state of the actor
+    * @param engine The reference to the current game engine
+    * @param delta  The time elapsed since the last update in milliseconds
+    */
+   public update(engine: Engine, delta: number) {
+      this._initialize(engine);
+      this.emit('preupdate', new PreUpdateEvent(engine, delta, this));
+
+      // Update action queue
+      this.actionQueue.update(delta);
+
+      // Update color only opacity
+      if (this.color) {
+         this.color.a = this.opacity;
+      }
+
+      // calculate changing opacity
+      if (this.previousOpacity !== this.opacity) {
+         this.previousOpacity = this.opacity;
+         this._opacityFx.opacity = this.opacity;
+         this._effectsDirty = true;
+      }
+
+      // Capture old values before integration step updates them
+      this.oldVel.setTo(this.vel.x, this.vel.y);
+      this.oldPos.setTo(this.pos.x, this.pos.y);
+
+      // Run Euler integration
+      this.integrate(delta);
+
+      // Update actor pipeline (movement, collision detection, event propagation, offscreen culling)
+      for (var trait of this.traits) {
+         trait.update(this, engine, delta);
+      }
+
+      // Update child actors
+      for (var i = 0; i < this.children.length; i++) {         
+         this.children[i].update(engine, delta);
```

Comment:
I moved the stats incrementing code to `Scene` instead, to remove the reference to `UIActor`.